zocadls/wrapper/generic.py

```diff
@@ -0,0 +1,34 @@
+from zocalo.wrapper import BaseWrapper
+import os
+import procrunner
+import logging
+
+logger = logging.getLogger('ProcessRegisterWrapper')
+
+class ProcessRegisterWrapper(BaseWrapper):
+
+    def run(self):
+        assert hasattr(self, 'recwrap'), "No recipewrapper object found"
+      
+        params = self.recwrap.recipe_step["job_parameters"]
+     
+        command = params["wrapped_commands"]
+        logger.info('Command: %s', ' '.join(command))
+        result = procrunner.run(command)
+        logger.info("Command successful, took %.1f seconds", result["runtime"])
+     
+        if "filename" in params:
+            self.record_result(params["filename"], "Result")
+    
+        if "logname" in params:
+            self.record_result(params["logname"], "Log")
+        return True
+
+    def record_result(self,path,type):
```

Comment:
Type isn't used in this function?

Reply:
Ha, of course, and I only test "Result" I dont have a test for "Log". It should be 'file_type': type.

I will fix this up.

Reply:
No worries, that's why we have pull requests and CI! :)